### ISSUE TITLE
IllegalArgumentException: Folder 'Trash' not found

### ISSUE THREAD

**Issue by DenBond7** [FlowCrypt/flowcrypt-android#527]
I think it's a bug from a non-Gmail account. Maybe `Trash` folder has a different flag (not `\Trash`) and that's why I can't identify this folder correctly.
```
[SUMMARY]
java.lang.IllegalArgumentException: Folder 'Trash' not found
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:


[STACK_TRACE]
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:
        at com.flowcrypt.email.util.exception.ExceptionUtil.handleError(ExceptionUtil.java:28)
        at com.flowcrypt.email.ui.activity.MessageDetailsActivity.onDeleteMsgClicked(MessageDetailsActivity.java:348)
        at com.flowcrypt.email.ui.activity.fragment.MessageDetailsFragment.runMsgAction(MessageDetailsFragment.java:495)
        at com.flowcrypt.email.ui.activity.fragment.MessageDetailsFragment.onOptionsItemSelected(MessageDetailsFragment.java:230)
        at androidx.fragment.app.Fragment.performOptionsItemSelected(Fragment.java:2587)
        at androidx.fragment.app.FragmentManagerImpl.dispatchOptionsItemSelected(FragmentManager.java:3368)
        at androidx.fragment.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:7356)
        at androidx.appcompat.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:196)
        at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
        at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
        at androidx.appcompat.app.ToolbarActionBar$2.onMenuItemClick(ToolbarActionBar.java:64)
        at androidx.appcompat.widget.Toolbar$1.onMenuItemClick(Toolbar.java:204)
        at androidx.appcompat.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected(ActionMenuView.java:781)
        at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:840)
        at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:158)
        at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:991)
        at androidx.appcompat.widget.ActionMenuView.invokeItem(ActionMenuView.java:2981)
        at androidx.appcompat.view.menu.ActionMenuItemView.onClick(ActionMenuItemView.java:151)
        at android.view.View.performClick(View.java:6612)
        at android.view.View.performClickInternal(View.java:6581)
        at android.view.View.access$3100(View.java:785)
        at android.view.View$PerformClick.run(View.java:25904)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:280)
        at android.app.ActivityThread.main(ActivityThread.java:6706)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: java.lang.IllegalArgumentException: Folder 'Trash' not found
        ... 28 more
java.lang.IllegalArgumentException: Folder 'Trash' not found
        at com.flowcrypt.email.ui.activity.MessageDetailsActivity.onDeleteMsgClicked(MessageDetailsActivity.java:348)
        at com.flowcrypt.email.ui.activity.fragment.MessageDetailsFragment.runMsgAction(MessageDetailsFragment.java:495)
        at com.flowcrypt.email.ui.activity.fragment.MessageDetailsFragment.onOptionsItemSelected(MessageDetailsFragment.java:230)
        at androidx.fragment.app.Fragment.performOptionsItemSelected(Fragment.java:2587)
        at androidx.fragment.app.FragmentManagerImpl.dispatchOptionsItemSelected(FragmentManager.java:3368)
        at androidx.fragment.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:7356)
        at androidx.appcompat.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:196)
        at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
        at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
        at androidx.appcompat.app.ToolbarActionBar$2.onMenuItemClick(ToolbarActionBar.java:64)
        at androidx.appcompat.widget.Toolbar$1.onMenuItemClick(Toolbar.java:204)
        at androidx.appcompat.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected(ActionMenuView.java:781)
        at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:840)
        at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:158)
        at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:991)
        at androidx.appcompat.widget.ActionMenuView.invokeItem(ActionMenuView.java:2981)
        at androidx.appcompat.view.menu.ActionMenuItemView.onClick(ActionMenuItemView.java:151)
        at android.view.View.performClick(View.java:6612)
        at android.view.View.performClickInternal(View.java:6581)
        at android.view.View.access$3100(View.java:785)
        at android.view.View$PerformClick.run(View.java:25904)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:280)
        at android.app.ActivityThread.main(ActivityThread.java:6706)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```
https://mail.google.com/mail/u/0/#search/label%3Awork-flowcrypt-crashes+label%3Aunread/FMfcgxwCgCQMGcNhSxjgQbppmSbjXBjz

**Comment by DenBond7** [comment]
https://mail.google.com/mail/u/0/#search/label%3Awork-flowcrypt-crashes+label%3Aunread/FMfcgxwCgCQMGcNhQmhtnkhLfVwgfnGd

`navtron.ro|Cannot move the message. The dest folder is null`

I see this user uses a custom email provider (maybe the company email provider(http://www.navtron.ro) with custom settings)

**Comment by DenBond7** [comment]
Fixed via #718
